lib/puppet/provider/user/hpux.rb

```diff
@@ -63,4 +75,43 @@ def password
       return ent.passwd
     end
   end
+
+  def trusted
+      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
+      if trusted_sys.chomp == "System is not trusted."
```

Comment:
Good catch!  I will go back to look over this and test if changing the 
test case order breaks anything.

On 1/2/2015 2:08 PM, Michael Smith wrote:
>
> In lib/puppet/provider/user/hpux.rb 
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22419055>:
>
> > @@ -63,4 +75,43 @@ def password
> >        return ent.passwd
> >      end
> >    end
> > +
> > +  def trusted
> > +      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
> > +      if trusted_sys.chomp == "System is not trusted."
>
> The default behavior if getprpw isn't found or returns an error is to 
> label the system "Trusted". Is that the desired default?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22419055>.
>